route/src/route_generator_worker.cpp

```diff
@@ -237,13 +258,19 @@ namespace route {
         lanelet::BasicPoint2d current_loc(msg->pose.position.x, msg->pose.position.y);
         // get dt ct from world model
         auto track = this->world_model_->routeTrackPos(current_loc);
+        auto via_lanelet_vector = lanelet::geometry::findNearest(world_model_->getMap()->laneletLayer, current_loc, 1);
+        auto current_lanelet = lanelet::ConstLanelet(via_lanelet_vector[0].second.constData());
+        auto lanelet_track = world_model_->trackPos(current_lanelet, current_loc);
+        ll_id_ = current_lanelet.id();
+        ll_crosstrack_distance_ = lanelet_track.crosstrack;
+        ll_downtrack_distance_ = lanelet_track.downtrack;
         current_crosstrack_distance_ = track.crosstrack;
         current_downtrack_distance_ = track.downtrack;
         // check if we left the seleted route by cross track error
         if(std::fabs(current_crosstrack_distance_) > cross_track_max_)
         {
             this->rs_worker_.on_route_event(RouteStateWorker::RouteEvent::LEFT_ROUTE);
-            publish_route_event(cav_msgs::RouteEvent::LEFT_ROUTE);
+            publish_route_event(cav_msgs::RouteEvent::ROUTE_DEPARTURE);
         }
         // check if we reached our destination be remaining down track distance
         if(current_downtrack_distance_ > world_model_->getRoute()->length2d() - down_track_target_range_)
```

Comment:
Line 278 is "ROUTE_COMPLETE" instead of "ROUTE_COMPLETED. Please do search all to make sure everything's been updated.